modules/tensor_mechanics/doc/content/modules/tensor_mechanics/NewMaterialSystem.md

```diff
@@ -0,0 +1,135 @@
+# New Material System for the Lagrangian Kernels
+
+## Purpose and Implementations
+
+The material system for the new Lagrangian kernels consists of:
+
+- The strain calculator for the new system, [`ComputeLagrangianStrain`](ComputeLagrangianStrain.md)
+- A stress update object, implemented as a derived class based on either
```

Comment:
I'd say 'stress calculator', rather than 'stress update' because we use the term 'stress update' in TM to describe the models that compute the inelastic strain to update the predicted stress. 

Also, we have been talking for a long time about getting rid of the `Compute` prefix on all the models (because it doesn't add anything to describe what it does). I would be in favor of dropping that in all your new models, but I'd consider that optional. Alternatively, we can just rename everything all at once sometime down the road.

Reply:
1. Sure,  I'll use "stress calculator"
2. How about we do everything at once sometime in the future :)

Reply:
Yeah, that's fine to kick it down the road.